JaceHensley/vscode-over-react-snippets
# AF-2920: add Dart 2 compatible snippets

## Description 

Several changes and additions to the over_react component boiler plate were required to reach parity between the current over_react transformer and the new Dart 2 over_react builder. This PR adds snippets that are compatible with the Dart 2 over_react builder. 

## Test

- open VS code, navigate to preferences and select ```User Snippets```
- select ```New Global Snippet file...```, name the file and paste in contents of ```snippets.json``` 
- you should now be able to invoke the snippets in any file with a ```.dart``` extension by calling the snippet prefix 

## Review 
@evanweible-wf  @corwinsheahan-wf @JaceHensley
